src/PlusImageProcessing/Testing/vtkPlusTransverseProcessEnhancerTest.cxx

```diff
@@ -36,10 +62,130 @@ int main(int argc, char **argv)
 
-  ///////////////
+  //check for various non-optional command line arguments
+  int commandCheckStatus = NULL;
+
+  if (inputFileName.empty())
+  {
+    std::cerr << "--input-seq-file not found!" << std::endl;
+    commandCheckStatus = EXIT_FAILURE;
+  }
+  if (outputFileName.empty())
+  {
+    std::cerr << "--output-seq-file not found!" << std::endl;
+    commandCheckStatus = EXIT_FAILURE;
+  }
+  if (configFileName.empty())
+  {
+    std::cerr << "--config-file not found!" << std::endl;
+    commandCheckStatus = EXIT_FAILURE;
+  }
+
+  if (commandCheckStatus == EXIT_FAILURE)
+  {
+    return EXIT_FAILURE;
+  }
 
-  // TODO : write code that tests the functionality of your tool here
   vtkSmartPointer<vtkPlusTransverseProcessEnhancer> enhancer = vtkSmartPointer<vtkPlusTransverseProcessEnhancer>::New();
 
+  // Read input sequence
+  vtkSmartPointer<vtkPlusTrackedFrameList> trackedFrameList = vtkSmartPointer<vtkPlusTrackedFrameList>::New();
+  if (trackedFrameList->ReadFromSequenceMetafile(inputFileName.c_str()) == PLUS_FAIL)
+  {
+    return EXIT_FAILURE;
+  }
+  vtkSmartPointer<vtkImageCast> castToUchar = vtkSmartPointer<vtkImageCast>::New();
+  castToUchar->SetOutputScalarTypeToUnsignedChar();
+
+  int numberOfFrames = trackedFrameList->GetNumberOfTrackedFrames();
+  std::cout << "Number of frames in input: " << numberOfFrames << std::endl;
+
+
+  //Check methods of vtkPlusTransverseProcessEnhancer for failure
+
+  enhancer->SetInputFrames(trackedFrameList);
+
+  //test the abillity to read to the config file
+  LOG_DEBUG("Reading config file...");
+  std::cout << "Reading Config file." << std::endl;
+  vtkSmartPointer<vtkXMLDataElement> configRootElement = vtkSmartPointer<vtkXMLDataElement>::New(); 
+  if (PlusXmlUtils::ReadDeviceSetConfigurationFromFile(configRootElement, configFileName.c_str()) == PLUS_FAIL)
+  {
+    LOG_ERROR("Unable to read configuration from file " << configFileName.c_str());
+    std::cout << "Failed to read Config file" << std::endl;
+
+    return EXIT_FAILURE;
+  }
+
+  vtkXMLDataElement* processorElement = configRootElement->FindNestedElementWithName("ScanConversion");
+  if (processorElement == NULL)
+  {
+    LOG_ERROR("Cannot find device set in XML tree element: " << ((*configRootElement).GetName()));
+    return PLUS_FAIL;
+  }
+
+  if (enhancer->ReadConfiguration(configRootElement) == PLUS_FAIL)
+  {
+    LOG_ERROR("Unable to read configuration from file " << configFileName.c_str());
+    std::cout << "Failed to read Config file" << std::endl;
+    return EXIT_FAILURE;
+  }
+  LOG_DEBUG("Reading config file finished.");
+  std::cout << "Done reading Config file." << std::endl;
+
+  //Process the frames for the input file
+  LOG_DEBUG("attempting to Process Frames.");
+  std::cout << "Processing Frame." << std::endl;
+  if (enhancer->Update() == PLUS_FAIL)
+  {
+    LOG_ERROR("Could not Process Frames.");
+    std::cout << "Failed processing frames." << std::endl;

```

Comment:
If you already log the error then there is no need for printing on cout.
LOG_ macros serve as a more structured and controllable way of printing to cout, cerr, and log files.